src/osd/ECBackend.cc

```diff
@@ -1012,7 +1014,23 @@ shard_to_read_map.find(from);
   assert(rop.in_progress.count(from));
   rop.in_progress.erase(from);
   if (!rop.in_progress.empty()) {
-    dout(10) << __func__ << " readop not complete: " << rop << dendl;
+    if (subread_all && !rop.is_recovery) {
+      int k = ec_impl->get_data_chunk_count();
+      for (map<hobject_t, read_result_t>::iterator i =
+            rop.complete.begin();
+          i != rop.complete.end();
+          ++i) {
+        if ( i->second.returned.front().get<2>().size() < k ) {
```

Comment:
@guangyy , i might completely wrong. but i think this only applies to jerasure. you might want to use `ec_impl->minimum_to_decode(want, have, &need)` instead. to be specific

1. prepare `want_to_read` using `ec_impl->get_chunk_mapping()`
1. pass `rop.complete` to `minimum_to_decode()` as `available_chunks` every time a sub read op returns.
1. if `-EIO` is returned, then "read all shards - readop not complete", otherwise, we can move on and consider the read op completed.
1. use the returned `need` for decoding if read op completes.